### PR TITLE
Add test case WALA-VERIFY-HOSTNAME-CHANGE

### PR DESCRIPTION
**Pull request by LiliDeng**
This case will rename hostname, and WALA will detect this change when Provisioning.MonitorHostName=y

Test results -
```
[LISAv2 Test Results Summary]
Test Run On           : 05/29/2020 13:43:18
ARM Image Under Test  : SUSE : sles-12-sp5 : gen1 : latest
Initial Kernel Version: 4.12.14-16.10-azure
Final Kernel Version  : 4.12.14-16.10-azure
Total Test Cases      : 2 (2 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 WALA                 WALA-VERIFY-FIREWALL-STATUS                                                       PASS                 0.57 
    2 WALA                 WALA-VERIFY-HOSTNAME-CHANGE                                                       PASS                 2.58 

[LISAv2 Test Results Summary]
Test Run On           : 05/29/2020 13:42:12
ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
Initial Kernel Version: 4.15.0-1082-azure
Final Kernel Version  : 4.15.0-1082-azure
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:5

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 WALA                 WALA-VERIFY-HOSTNAME-CHANGE                                                       PASS                 3.32 

[LISAv2 Test Results Summary]
Test Run On           : 05/29/2020 13:32:19
ARM Image Under Test  : CoreOS : CoreOS : Stable : Latest
Initial Kernel Version: 4.19.123-coreos
Final Kernel Version  : 4.19.123-coreos
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:5

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 WALA                 WALA-VERIFY-HOSTNAME-CHANGE                                                       PASS                 2.59 
```

### CODE REVIEW

File: Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py [hunk 1]
```diff
@@ -17,8 +17,9 @@
 EXECUTION_USER = "root"
 WIRESERVER_ENDPOINT_FILE = '/var/lib/waagent/WireServerEndpoint'
 VERSIONS_PATH = '/?comp=versions'
-AGENT_CONFIG_FILE = '/etc/waagent.conf'
 OS_ENABLE_FIREWALL_RX = r'OS.EnableFirewall\s*=\s*(\S+)'
+AGENT_CONFIG_FILE = Run("find / -name waagent.conf")
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py:21]: Why do we need this change? waagent.conf location is static.

File: XML/TestCases/FunctionalTests-WALA.xml [hunk 2]
```diff
@@ -128,4 +128,18 @@
         <Tags>wala</Tags>
         <Priority>2</Priority>
     </test>
+    <test>
+        <TestName>WALA-VERIFY-HOSTNAME-CHANGE</TestName>
+        <testScript>WALA-VERIFY-HOSTNAME-CHANGE.py</testScript>
```
Comment (juhlee-microsoft) [inline XML/TestCases/FunctionalTests-WALA.xml]: This python script has many lines in common with WALA-VERIFY-HOSTNAME.py. Can we use a single script with two different parameters?

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py (hunk 1 above)
Comment (LiliDeng) [reply Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py:21]: it has different path in coreos

Comment (juhlee-microsoft) [reply Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py:21]: Should we open and ask waagent team fix the path?I can merge this way but we have a few other scripts with the old way.

File: XML/TestCases/FunctionalTests-WALA.xml (hunk 2 above)
Comment (LiliDeng) [reply XML/TestCases/FunctionalTests-WALA.xml]: sure, thanks Ju, I'd like move the common functions to the library file.

File: Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/WALA-VERIFY-FIREWALL-STATUS.py:21]: I see it's end-of-life, June 2020. No need to change then.

File: XML/TestCases/FunctionalTests-WALA.xml (hunk 2 above)
Comment (juhlee-microsoft) [reply XML/TestCases/FunctionalTests-WALA.xml]: If this is ready for the review, please post the label 'Ready-For-Review'

**Comment by juhlee-microsoft** [comment]
@LiliDeng Are you still working on this PR?

**Comment by harz566** [comment]
@LiliDeng   Hello, please kindly note, LISA has been refactored in its xml definition.  Generally,  if you have `<AdditionalHWConfig>` defined, you need to replace `<AdditionalHWConfig>`  with `<SetupConfig>`, otherwise, you need create a new XML Tag `<SetupConfig>`,  and move any of these xml Tags (`<SetupType>, <OverrideVMSize>, <ARMImageName>, <TestLocation>, <Networking>, <DiskType>, <ImageType>, <OSType>, <OsVHD>`)  as child element of `<SetupConfig>`.   Some of those Tags are optional, please reference to the existing xml definition from master branch.   Hope that doesn't bring much inconvenience to you. Thanks!

**Comment by juhlee-microsoft** [comment]
@LiliDeng Is PR open in more than 1 month? are you still working on it?

**Comment by LiliDeng** [comment]
> @LiliDeng Is PR open in more than 1 month? are you still working on it?

Work on today.

File: XML/TestCases/FunctionalTests-WALA.xml (hunk 2 above)
Comment (LiliDeng) [reply XML/TestCases/FunctionalTests-WALA.xml]: @juhlee-microsoft  I double check, only CheckHostName function is duplicated, and It is not suitable to move this into common library.

**Review by juhlee-microsoft — APPROVED** [review]

**Review by lubaihua33 — APPROVED** [review]

**Review by harz566 — APPROVED** [review]

**Pull request merged**